index.js

```diff
@@ -136,43 +160,74 @@ function initialize() {
         if (err) return cb(err)
         var json = gist.data
         if (!json.files || !json.files['index.js']) return cb({error: 'no index.js in this gist', json: json})
-        var code = json.files['index.js'].content
+        code.head = json.files['page-head.html'].content
+        code.body = json.files['page-body.html'].content
+        code.bundle = json.files['index.js'].content
+        code.meta = json.files['package.json'].content
         var pj = json.files['package.json']
         if (pj) {
           try { pj = JSON.parse(pj.content) }
           catch (e) { pj = false }
           if (pj) packagejson.dependencies = pj.dependencies
         }
-        codeMD5 = md5(code)
-        cb(false, code)
+        codeMD5 = md5(code.bundle)
+        invokeCallback()
       })
     }
 
-    var stored = localStorage.getItem('code')
-    if (stored) return cb(false, stored)
-
-    var defaultCode = document.querySelector('#template').innerText
-    cb(false, defaultCode)
+    code.bundle = localStorage.getItem('bundleCode') ||
+      document.querySelector('#bundle-template').innerText
+    code.head = localStorage.getItem('headCode') || '';
+    code.body = localStorage.getItem('bodyCode') || '';
+    code.meta = localStorage.getItem('metaCode') || '';
+    invokeCallback();
   }
 
   loadCode(function(err, code) {
     if (err) return alert(JSON.stringify(err))
 
-    var editor = jsEditor({
+    // javascript editors
+    var bundleEditor = jsEditor({
       container: editorEl,
       lineWrapping: true
     })
+    bundleEditor.name = 'bundle';
+
+    var metaEditor = jsEditor({
+      value: JSON.stringify(packagejson),
```

Comment:
IMO this shold be `JSON.stringify(packagejson, null, '  ')` so that the JSON gets pretty printed